src/swmmanywhere/graphfcns/design_graphfcns.py

```diff
@@ -15,102 +15,129 @@
-        average_depth = (depth + chamber_floor) / 2
-        V = edge_length * (diam + 0.3) * (average_depth + 0.1)
-        cost = 1.32 / 2000 * (9579.31 * diam**0.5737 + 1163.77 * V**1.31)
-        v_feasibility = max(hydraulic_design.min_v - v, 0) + max(
-            v - hydraulic_design.max_v, 0
-        )
-        fr_feasibility = max(filling_ratio - hydraulic_design.max_fr, 0)
-        """
-        TODO shear stress... got confused here
-        density = 1000
-        dyn_visc = 0.001
-        hydraulic_diameter = 4 * (A * filling_ratio**2) / \
-            (np.pi * diam * filling_ratio)
-        Re = density * v * 2 * (diam / 4) * (filling_ratio ** 2) / dyn_visc
-        fd = 64 / Re
-        shear_stress = fd * density * v**2 / fd
-        shear_feasibility = max(min_shear - shear_stress, 0)
-        """
-        slope = (chamber_floor - (ds_elevation - depth)) / edge_length
-        pipes.append(
-            {
-                "diam": diam,
-                "depth": depth,
-                "slope": slope,
-                "v": v,
-                "fr": filling_ratio,
-                # 'tau' : shear_stress,
-                "cost": cost,
-                "v_feasibility": v_feasibility,
-                "fr_feasibility": fr_feasibility,
-                "surcharge_feasibility": surcharge_feasibility,
-                # 'shear_feasibility' : shear_feasibility
-            }
-        )
+    fr_feasibility = max(filling_ratio - hydraulic_design.max_fr, 0)
+    """
+    TODO shear stress... got confused here
+    density = 1000
+    dyn_visc = 0.001
+    hydraulic_diameter = 4 * (A * filling_ratio**2) / \
+        (np.pi * diam * filling_ratio)
+    Re = density * v * 2 * (diam / 4) * (filling_ratio ** 2) / dyn_visc
+    fd = 64 / Re
+    shear_stress = fd * density * v**2 / fd
+    shear_feasibility = max(min_shear - shear_stress, 0)
+    """
+    slope = (chamber_floor - (ds_elevation - depth)) / edge_length
+    return {
+        "diameter": diam,
+        "depth": depth,
+        "slope": slope,
+        "v": v,
+        "fr": filling_ratio,
+        # 'tau' : shear_stress,
+        "cost_usd": cost,
+        "v_feasibility": v_feasibility,
+        "fr_feasibility": fr_feasibility,
+        "surcharge_feasibility": surcharge_feasibility,
+        # 'shear_feasibility' : shear_feasibility
+    }
+
+
+def select_design(pipes_df: pd.DataFrame) -> dict[Hashable, float]:
+    """Select the ideal design from the dataframe.
+
+    This function selects the ideal design from the dataframe by sorting the
+    dataframe based on the feasibility parameters and cost. It returns the
+    diameter, depth, and cost of the ideal design.
 
-    pipes_df = pd.DataFrame(pipes).dropna()
+    Args:
+        pipes_df (pd.DataFrame): A dataframe containing the designs and their
+            parameters
+
+    Returns:
+        dict: A dictionary containing the ideal design
+    """
     if pipes_df.shape[0] > 0:
```

Comment:
I would do this the other way around, for clarity:

```suggestion
    if pipes_df.shape[0] <= 0:
        raise ValueError("No non nan pipes designed. Shouldn't happen.")
```

And then the rest of the code outside of an `else` block.